tests/e2e/lib/diskutils

```diff
@@ -0,0 +1,108 @@
+
+  for disk in $disks_arr; do
+     if [[ ${disk} == "vda" || \
+           $(echo "${disk_table}" | grep -c "${disk}" ) -eq 1 && ${disk} != "zram0" ]];then
+         new_part="$( (echo n; echo p; echo ; echo ; echo ; echo w) | fdisk "/dev/${disk}")"
+         part_id="$(echo "${new_part}" | grep -Po "new partition \K([0-9])")"
+         DISK="${disk}"
+         if [[ $(echo "${DISK}" | grep -c nvme) -eq 1 ]]; then
+           PART_ID="p${part_id}"
+           info_message "select_disk_to_partition, found /dev/${DISK}${PART_ID}"
+           info_message "=============================="
+           break;
+         fi
+      fi
+      if [ -e /sys/devices/soc0/machine ]; then
+        if grep -qi "${QC_SOC}" /sys/devices/soc0/machine; then
+           if [[ ${disk} == "sde" ]]; then
+             DISK=${disk}
+             part=$(echo "${disk_table}" | grep "G" | grep "sde[0-9]." | grep -v "/" | cut -d" " -f1 )
+             PART_ID=${part:3}
+             break;
+           fi
+        fi
+      fi
+  done
+}
+
+create_qm_var_part() {
+  ###########################################################################
+  # Description:                                                            #
+  #  Create relevant filesystem                                             #
+  #    support fstype, non ostree, xfs (c9s vm), ext4(soc)                  #
+  # ARGUMENTS: should be be exported by calling script                      #
+  #   DEVICE,  set free /dev/ device                                        #
+  #   PART_ID, set device partition                                         #
+  ###########################################################################
+
+  select_disk_to_partition
+
+  local slash_var
+  slash_var="/var"
+
+  # Check is SoC aboot exist, else regular image
+  if [ -e /sys/devices/soc0/machine ]; then
+        if grep -qi "${QC_SOC}" /sys/devices/soc0/machine; then
+          mkfs.ext4 "/dev/${DISK}${PART_ID}"
+          if_error_exit "Error: mkfs.ext4 /dev/${DISK}${PART_ID} failed on ${QC_SOC}"
+          info_message "Create_qm_disks, prepare SoC ext4 fs"
+          info_message "=============================="
+        fi
+  else
+  # VM use case
+       mkfs.xfs "/dev/${DISK}${PART_ID}"
+       if_error_exit "Error: mkfs.xfs /dev/${DISK}${PART_ID} failed on VM"
+       info_message "Create_qm_disks, prepare regular xfs fs"
+       info_message "=============================="
+
+  fi
+
+  mkdir -p /new_var
+  mount "/dev/${DISK}${PART_ID}" /new_var
+  rsync -aqxXP /var/* /new_var
+  if_error_exit "Error: rsync failed"
+  systemctl stop var-lib-nfs-rpc_pipefs.mount
+  umount /new_var
+  rm -rf "${slash_var:?}"/*
+  info_message "Create_qm_disks, prepare and mount ${slash_var}"
+  info_message "=============================="
+  mount "/dev/${DISK}${PART_ID}" "${slash_var}"
+  if_error_exit "Error: mount /dev/${DISK}${PART_ID} on ${slash_var} failed"
+  systemctl start var-lib-nfs-rpc_pipefs.mount
```

Comment:
should we check for error in this one too ?

Reply:
Ack